ng-dev/misc/new-main-branch/cli.ts

```diff
@@ -0,0 +1,115 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {assertValidGithubConfig, getConfig} from '../../utils/config';
+import {error, info, promptConfirm, red, warn, yellow} from '../../utils/console';
+import {getRemotesForRepo, isAngularOwnedRemote} from './remotes';
+
+import {CommandModule} from 'yargs';
+import {GitClient} from '../../utils/git/git-client';
+
+async function handler() {
+  const git = GitClient.get();
+  const config = getConfig([assertValidGithubConfig]);
+  const repoSlug = `${config.github.owner}/${config.github.name}`;
+
+  if (config.github.mainBranchName !== 'main') {
+    error(red('Current project is not part of the default branch renaming.'));
+    return;
+  }
+
+  if (git.hasUncommittedChanges()) {
+    error(red('There are uncommitted changes. Unable to switch to new main branch. Aborting..'));
+    return;
+  }
+
+  if (!hasLocalBranch(git, 'master')) {
+    error(red('Local repository does not have a local branch named `master`. Aborting..'));
+    return;
+  }
+
+  if (hasLocalBranch(git, 'main')) {
+    error(red('Local repository already has a branch named `main`. Aborting..'));
+    return;
+  }
+
+  const remotes = getRemotesForRepo(git);
+  const angularRemotes = Array.from(remotes.entries()).filter((r) => isAngularOwnedRemote(r[1]));
+  const angularRemoteNames = angularRemotes.map(([name]) => name);
+  const primaryRemoteName = angularRemoteNames[0];
```

Comment:
What makes this the primary remote?  Aren't the remotes just in alphabetical order?

Reply:
added a comment. Usually there is only one in a local repository, but if there are more we can just pick _any_